### DISCUSSION TITLE
Why it only allow single image uplad?

### DISCUSSION THREAD

**Discussion by hatimmakki** [inventare/django-image-uploader-widget#97] [General]
Hi

I have this setup, but the user can only upload 1 file. How to enable multiple image uploads?

Thanks

```
class EventForm(forms.ModelForm):
    #  upload multiple images using ImageUploaderWidget from image_uploader_widget package
    images = forms.ImageField(widget=ImageUploaderWidget(), required=False)
    
    class Meta:
        model = Event
        fields = [
            'title', 
            'description', 
            'start_date', 
            'end_date', 
            'location', 
            'price', 
            'currency', 
            'max_tickets', 
            'is_published', 
            'target_audiences',
            'images'
        ]
        

```

**Comment by EduardoJM** [comment]
Hi, 

The django ImageField is, in the concept, an CharField that saves the path to file into the configured storage. One of the consequences of this behaviour of the ImageField, is: its accepts only one image (with widget or without it).

To allow multiple images upload, in the case of django-admin, is create an another Model to stores image and uses (1-n relation) and uses the custom `ImageUploaderInline`, like this: https://inventare.github.io/django-image-uploader-widget/docs/inline_admin/tutorial/#using-the-inline-editor

To allow multiple images upload outside of django-admin, currently i have no idea of how to do this using this library (and i create an issue #98 to investigate and create this feature).

**Comment by EduardoJM** [comment]
For `PostgreSQL`, is possible, currently, to use the `ArrayField` supported by the custom field:

https://inventare.github.io/django-image-uploader-widget/array_field/tutorial/
